sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java

```diff
@@ -1077,19 +1078,27 @@ static SearchRequest createSearchRequest(String searchText, SearchOptions option
     static String createSearchRequestAnswers(SearchOptions searchOptions) {
         QueryAnswerType answer = searchOptions.getAnswers();
         Integer answersCount = searchOptions.getAnswersCount();
+        Double answerThreshold = searchOptions.getAnswerThreshold();
 
         // No answer has been defined.
         if (answer == null) {
             return null;
         }
 
-        // No count, just send the QueryAnswer.
-        if (answersCount == null) {
-            return answer.toString();
+        String answerString = answer.toString();
+
+        if (answersCount != null && answerThreshold != null) {
+            return answerString + "|count-" + answersCount + ",threshold-" + answerThreshold;
+        }
+
+        if (answersCount != null) {
+            answerString = answerString + "|count-" + answersCount;
         }
 
-        // Answer and count, format it as the service expects.
-        return answer + "|count-" + answersCount;
+        if (answerThreshold != null) {
+            answerString = answerString + "|threshold-" + answerThreshold;
+        }
+        return answerString;
```

Comment:
I'd make this one big if/else if/else block where all conditional blocks return

Reply:
Done